portfolio/src/test/java/com/google/sps/TaskQueueTest.java

```diff
@@ -66,7 +65,153 @@ public void checkCorrectSortingShortestTaskFirst() {
 
-    for (int i = 0; i < expected.size(); i++) {
+    int i = 0;
+    while (i++ < expected.size()) {
+      actual.add(taskQueue.peek());
+      taskQueue.remove();
+    }
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  /**
+   * Ensure that the PriorityQueue is correctly sorting the tasks by Priority using the
+   * HIGHEST_PRIORITY_FIRST comparator.
+   */
+  @Test
+  public void checkCorrectPrioritySortingHighestPriorityFirst() {
+    Task task1 =
+        new Task("Task 1", "First task", DURATION_30_MINUTES, SchedulerTestUtil.PRIORITY_ONE);
+    Task task2 =
+        new Task("Task 2", "Second task", DURATION_30_MINUTES, SchedulerTestUtil.PRIORITY_FIVE);
+    Task task3 =
+        new Task("Task 3", "Third task", DURATION_30_MINUTES, SchedulerTestUtil.PRIORITY_TWO);
+    Task task4 =
+        new Task("Task 4", "Fourth task", DURATION_30_MINUTES, SchedulerTestUtil.PRIORITY_FOUR);
+    Task task5 =
+        new Task("Task 5", "Fifth task", DURATION_30_MINUTES, SchedulerTestUtil.PRIORITY_THREE);
+
+    List<Task> taskList = Arrays.asList(task3, task1, task2, task5, task4);
+
+    List<Task> expected = Arrays.asList(task2, task4, task5, task3, task1);
+    List<Task> actual = new ArrayList<Task>();
+
+    TaskQueue taskQueue = new TaskQueue(taskList, HIGHEST_PRIORITY_FIRST);
+
+    int i = 0;
+    while (i++ < expected.size()) {
+      actual.add(taskQueue.peek());
+      taskQueue.remove();
+    }
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  /**
+   * Ensure that the PriorityQueue is correctly sorting the tasks from shortest to longest using the
+   * HIGHEST_PRIORITY_FIRST comparator.
+   */
+  @Test
+  public void checkCorrectSamePrioritySortingHighestPriorityFirst() {
+    Task task1 =
+        new Task(
+            "Task 1",
+            "First task",
+            SchedulerTestUtil.DURATION_5_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+    Task task2 =
+        new Task(
+            "Task 2",
+            "Second task",
+            SchedulerTestUtil.DURATION_30_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+    Task task3 =
+        new Task(
+            "Task 3",
+            "Third task",
+            SchedulerTestUtil.DURATION_10_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+    Task task4 =
+        new Task(
+            "Task 4",
+            "Fourth task",
+            SchedulerTestUtil.DURATION_15_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+    Task task5 =
+        new Task(
+            "Task 5",
+            "Fifth task",
+            SchedulerTestUtil.DURATION_20_MINUTES,
+            SchedulerTestUtil.PRIORITY_ONE);
+
+    List<Task> taskList = Arrays.asList(task3, task1, task2, task5, task4);
+
+    List<Task> expected = Arrays.asList(task1, task3, task4, task5, task2);
+    List<Task> actual = new ArrayList<Task>();
+
+    TaskQueue taskQueue = new TaskQueue(taskList, HIGHEST_PRIORITY_FIRST);
+
+    int i = 0;
+    while (i++ < expected.size()) {
```

Comment:
We can introduce a method to taskQueue like isEmpty(), and then here do,

while (!taskQueue.isEmpty()) {

Reply:
Done.